### PR TITLE
The default template to use doc_values wherever possible

### LINKED ISSUE #275: Improve default template
I suggest the following:
* Use "index": "not_analyzed" on all string fields (except message?)
* Use "doc_values": true on all string fields

### PR DESCRIPTION
**Pull request by untergeek**
Includes mapping examples for other numeric types.

Types other than long and double will not be used unless manually
mapped by the user.

fixes #275

### CODE REVIEW

**Comment by jordansissel** [comment]
"resubmit ..." is a confusing commit message, can you reword it to describe the change being made? (for posterity)

**Comment by jordansissel** [comment]
The changelog entry "the default template to use doc_values wherever possible" seems like a good candidate for the git commit subject line :)

**Comment by untergeek** [comment]
Roger that. :+1:

**Comment by andrewvc** [comment]
Tests pass. Aside from Jordan's comment, LGTM!

**Comment by untergeek** [comment]
I'll call it good, then as the major review for this was in #276
